rest-api/README.md

```diff
@@ -0,0 +1,329 @@
+| ------------------ | --------------------------------- |
+| `200: OK`          | `{ data: TbdexMessage<Quote>[] }` |
+| `400: Bad Request` | `{ errors: Error[] }`             |
+
+### Response Body
+> [!IMPORTANT]
+> See Quote structure [here](../README.md#quote)
+
+## Submit Order
+
+### Endpoint
+`POST /threads/:thread_id/order`
+
+### Order Request Body
+> [!IMPORTANT]
+> See Order structure [here](../README.md#order)
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description                                                             |
+| ------ | ----------------------------------------------------------------------- |
+| 400    | Failed Signature Check                                                  |
+| 404    | Thread not found                                                        |
+| 409    | Order already exists                                                    |
+| 409    | Order not allowed (e.g. bc thread was closed or bc no quote exists yet) |
+
+## Submit Close
+
+### Description
+Closes the thread. Indicates that Alice is no longer interested
+
+### Endpoint
+`POST /threads/:thread_id/close`
+
+### Request Body
+> [!IMPORTANT]
+> See Close structure [here](../README.md#close)
+
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `202: Accepted`    | N/A                   |
+| `400: Bad Request` | `{ errors: Error[] }` |
+
+### Errors
+| Status | Description            |
+| ------ | ---------------------- |
+| 400    | Failed Signature Check |
+| 404    | Thread not found       |
+| 409    | Close not allowed      |
+
+---
+
+## Get Exchange (DESCOPED)
+
+### Description
+Retrieves the messages specified by ID and messageType
+
+### Authentication
+Uses DID authn via Bearer token in header.
+
+### Endpoint
+`GET /exchanges/:id`
+
+
+### Query Params
+| Param         | Description                   |
+| ------------- | ----------------------------- |
+| `messageType` | filters the messages returned  |
+
+
+### Response
+
+| Status             | Body                       |
+| ------------------ | -------------------------- |
+| `200: OK`          | `{ data: TbdexMessage[] }` |
+| `400: Bad Request` | `{ errors: Error[] }`      |
+| `404: Not Found`   | N/A                        |
+| `403: Forbidden`   | N/A                        |
+
+---
+
+## List Exchanges
+
+### Description
+Returns a map of Exchange ID to array of tbdex messages forming the exchange
+
+### Authentication
+Uses DID authn via Bearer token in header.
+
+### Endpoint
+`GET /exchanges`
+
+### Response
+| Status             | Body                  |
+| ------------------ | --------------------- |
+| `200: OK.     `    | `{ data: { string: tbdexMessage[] } }`   | 
```

Comment:
@phoebe-lew based on what you wrote here, if i call `GET /exchanges?ids=1,2,3`, should i expect the following?

```
{
    1 : [rfq1, quote1, order1],
    2 : [rfq2], 
    3 : [rfq3, quote3, order3, orderstatus3]
}
```

this would work. if it's easier, returning 2d array could also work
```
[
    [rfq1, quote1, order1],
    [rfq2], 
    [rfq3, quote3, order3, orderstatus3]
]
```

right now the didpay implementation assumes `tbdexMessage[][]` type as the response from this endpoint. but it can easily changed if need be to the `Map<string, tbdexMessage[]>`